### PR TITLE
Added support for context

### PR DESCRIPTION
**Pull request by Drioastorga**
Added support for context.

Current version allows to switch context between omp, native and sequential back-ends. Using TBB as back-end uses sequential implementation.

### CODE REVIEW

File: include/common/context.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+*/
+#ifndef GRPPI_COMMON_CONTEXT_H
+#define GRPPI_COMMON_CONTEXT_H
+
+#include <type_traits>
+
+namespace grppi {
+
+/**
+\brief Representation of farm pattern.
```
Comment (jdgarciauc3m) [inline include/common/context.h]: Wrong documentation

File: include/common/context.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  /**
+  \brief Invokes the trasnformer of the farm over a data item.
+  */
+  template <typename I>
+  auto operator()(I && item) const {
+    return transformer_(std::forward<I>(item));
+  }
+
+private:
+  int cardinality_;
```
Comment (jdgarciauc3m) [inline include/common/context.h]: Probably this member variable is not needed. Isn't it?

File: include/context.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+/**
+\brief Define a new \ref md_context on a data stream 
+that can be composed in other streaming patterns.
+\tparam Execution Execution policy type.
+\tparam Transformer Callable type for the transformation operation.
+\param ex Execution policy object.
+\param transform_op Transformer operation.
+*/
+template <typename ExecutionPolicy, typename Transformer>
+auto context(ExecutionPolicy & ex, Transformer && transform_op)
```
Comment (jdgarciauc3m) [inline include/context.h]: Find a better name.
Candidates: `run`, `run_with`, `run_at`

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
Address coments

**Comment by Drioastorga** [comment]
Renamed context interface to 'run_with'. 
Reviewed documentation.

**Review by jdgarciauc3m — APPROVED** [review]
OK. But I need to add some example application. To be handled as separate issue.

**Pull request merged**